starter-code/scripts/models/article.js

```diff
@@ -89,13 +102,27 @@ Article.numWordsByAuthor = function() {
     return Article.allAuthors().map(function(currentAuthor) {
       return {
         name: currentAuthor,
-        numWords: // someCollection.filter(function(curArticle) {
+        numWords: Article.allArticles.map(function(currentArticle) {
+          if (currentArticle.author === currentAuthor) {
+            return currentArticle.body.match(/\w+/g).length
+          }
+          else {
+            return null
+          }
+        })
+        .reduce(function(acc, current) {
+          return acc + current
+        }, 0)
+      }
+    })
+          // someCollection.filter(function(curArticle) {
           // what do we return here to check for matching authors?
         // .map() to return the author's word count for each article body (you may split or regexp)
         // .reduce() to squash this array into one big number, per author.
-      }
-    });
+      // }
 }
 
 module.Article = Article;
 })(window);
+
+Article.numWordsAll();
```

Comment:
You don't need to call this because, when adminPage.js is loaded via the script tag, Article.fetchAll is called and articleView.renderAdminPage is passed in as an argument. articleView.renderAdminPage will end up being invoked inside of Article.fetchAll and will itself call Article.numWordsAll.